### ISSUE TITLE
NPE when matching JSON

### ISSUE THREAD

**Issue by mantasindrasius** [etorreborre/specs2#617]
We are using specs2 version 3.8.6.

When matching a JSON containing an irrelevant null value, getting an NPE with the stacktrace below.

```scala
"match smth in summary" in {
      """{
        |  "b": {"a": 2,"c":null}
        |}""".stripMargin must /("b" -> /("a" -> 2))
    }
```

No underlying matcher is asserting agains "b" -> "c", but when removed, the test starts passing magically.

Stacktrace:
```
java.lang.NullPointerException: null
java.lang.NullPointerException
	at org.specs2.json.JSONFormat$$anonfun$1.apply(JSONType.scala:57)
	at org.specs2.json.JSONFormat$$anonfun$1.apply(JSONType.scala:53)
	at org.specs2.json.JSONObject$$anonfun$toString$1.apply(JSONType.scala:95)
	at org.specs2.json.JSONObject$$anonfun$toString$1.apply(JSONType.scala:95)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.HashMap$HashMap1.foreach(HashMap.scala:221)
	at scala.collection.immutable.HashMap$HashTrieMap.foreach(HashMap.scala:428)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.specs2.json.JSONObject.toString(JSONType.scala:95)
	at org.specs2.json.JSONFormat$$anonfun$1.apply(JSONType.scala:55)
	at org.specs2.json.JSONFormat$$anonfun$1.apply(JSONType.scala:53)
	at scala.collection.immutable.List.map(List.scala:284)
	at org.specs2.json.JSONArray.toString(JSONType.scala:105)
	at org.specs2.json.JSONFormat$$anonfun$1.apply(JSONType.scala:56)
	at org.specs2.json.JSONFormat$$anonfun$1.apply(JSONType.scala:53)
	at org.specs2.json.JSONObject$$anonfun$toString$1.apply(JSONType.scala:95)
	at org.specs2.json.JSONObject$$anonfun$toString$1.apply(JSONType.scala:95)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.Map$Map2.foreach(Map.scala:137)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.specs2.json.JSONObject.toString(JSONType.scala:95)
	at org.specs2.json.JSONType.toString(JSONType.scala:30)
	at java.lang.String.valueOf(String.java:2994)
	at java.lang.StringBuilder.append(StringBuilder.java:131)
	at scala.StringContext.standardInterpolator(StringContext.scala:125)
	at scala.StringContext.s(StringContext.scala:95)
	at org.specs2.matcher.JsonBaseMatchers$JsonMatcher$$anonfun$showMap$1.apply(JsonMatchers.scala:117)
	at org.specs2.matcher.JsonBaseMatchers$JsonMatcher$$anonfun$showMap$1.apply(JsonMatchers.scala:117)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.immutable.Map$Map2.foreach(Map.scala:137)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.specs2.matcher.JsonBaseMatchers$JsonMatcher.showMap(JsonMatchers.scala:117)
	at org.specs2.matcher.JsonBaseMatchers$JsonMatcher.org$specs2$matcher$JsonBaseMatchers$JsonMatcher$$selectorNotFound(JsonMatchers.scala:111)
	at org.specs2.matcher.JsonBaseMatchers$JsonMatcher.org$specs2$matcher$JsonBaseMatchers$JsonMatcher$$find(JsonMatchers.scala:80)
	at org.specs2.matcher.JsonBaseMatchers$JsonMatcher.apply(JsonMatchers.scala:33)
	at org.specs2.matcher.Matcher$$anon$4$$anonfun$apply$5.apply(Matcher.scala:138)
	at org.specs2.matcher.Matcher$$anon$4$$anonfun$apply$5.apply(Matcher.scala:138)
	at org.specs2.matcher.AndMatch.m2$lzycompute(MatchResult.scala:213)
	at org.specs2.matcher.AndMatch.m2(MatchResult.scala:213)
	at org.specs2.matcher.AndMatch.evaluate(MatchResult.scala:218)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator$$anonfun$and$2.apply(MatchResultCombinators.scala:49)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator$$anonfun$and$2.apply(MatchResultCombinators.scala:49)
	at scala.util.Either.fold(Either.scala:99)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator.and(MatchResultCombinators.scala:47)
	at org.specs2.matcher.Matcher$$anon$4.apply(Matcher.scala:138)
	at org.specs2.matcher.Matcher$$anon$1.apply(Matcher.scala:122)
	at org.specs2.matcher.Expectable.applyMatcher(Expectable.scala:50)
	at org.specs2.matcher.MustThrownExpectationsCreation$$anon$4.applyMatcher(MustExpectations.scala:67)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator$$anonfun$and$6$$anonfun$apply$9.apply(MatchResultCombinators.scala:71)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator$$anonfun$and$6$$anonfun$apply$9.apply(MatchResultCombinators.scala:71)
	at org.specs2.matcher.AndMatch.m2$lzycompute(MatchResult.scala:213)
	at org.specs2.matcher.AndMatch.m2(MatchResult.scala:213)
	at org.specs2.matcher.AndMatch.evaluate(MatchResult.scala:218)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator$$anonfun$and$6.apply(MatchResultCombinators.scala:71)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator$$anonfun$and$6.apply(MatchResultCombinators.scala:71)
	at scala.util.Either.fold(Either.scala:99)
	at org.specs2.matcher.MatchResultLogicalCombinators$MatchResultCombinator.and(MatchResultCombinators.scala:68)
	at com.wix.woa.NotificationsAPIE2E$$anonfun$1$$anonfun$apply$7$$anon$3.<init>(NotificationsAPIE2E.scala:71)
	at com.wix.woa.NotificationsAPIE2E$$anonfun$1$$anonfun$apply$7.apply(NotificationsAPIE2E.scala:60)
	at com.wix.woa.NotificationsAPIE2E$$anonfun$1$$anonfun$apply$7.apply(NotificationsAPIE2E.scala:60)
	at org.specs2.matcher.Scope$$anon$3$$anonfun$asResult$1.apply(ThrownExpectations.scala:151)
	at org.specs2.matcher.Scope$$anon$3$$anonfun$asResult$1.apply(ThrownExpectations.scala:151)
	at org.specs2.execute.ResultExecution$class.execute(ResultExecution.scala:23)
	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118)
	at org.specs2.execute.Result$$anon$10.asResult(Result.scala:229)
	at org.specs2.execute.AsResult$.apply(AsResult.scala:25)
	at org.specs2.matcher.Scope$$anon$3.asResult(ThrownExpectations.scala:151)
	at org.specs2.execute.AsResult$.apply(AsResult.scala:25)
	at org.specs2.specification.core.Execution$$anonfun$result$2.apply(Execution.scala:129)
	at org.specs2.specification.core.Execution$$anonfun$result$2.apply(Execution.scala:129)
	at org.specs2.specification.core.Execution$$anonfun$withEnv$1$$anonfun$apply$5.apply(Execution.scala:132)
	at org.specs2.execute.ResultExecution$class.execute(ResultExecution.scala:23)
	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118)
	at org.specs2.execute.Result$$anon$10.asResult(Result.scala:229)
	at org.specs2.execute.AsResult$.apply(AsResult.scala:25)
	at org.specs2.specification.core.Execution$$anonfun$withEnv$1.apply(Execution.scala:132)
	at org.specs2.specification.core.Execution$$anonfun$withEnv$1.apply(Execution.scala:132)
	at org.specs2.specification.create.ContextualFragmentFactory$$anonfun$example$3$$anonfun$apply$1$$anonfun$apply$2.apply(FragmentFactory.scala:106)
	at org.specs2.specification.create.ContextualFragmentFactory$$anonfun$example$3$$anonfun$apply$1$$anonfun$apply$2.apply(FragmentFactory.scala:106)
	at org.specs2.execute.ResultExecution$class.execute(ResultExecution.scala:23)
	at org.specs2.execute.ResultExecution$.execute(ResultExecution.scala:118)
	at org.specs2.execute.Result$$anon$10.asResult(Result.scala:229)
	at org.specs2.execute.AsResult$.apply(AsResult.scala:25)
	at com.wixpress.framework.test.env.GlobalTestEnvSupport$class.around(GlobalTestEnvSupport.scala:34)
	at com.wix.woa.BaseSessionServerIT.around(BaseSessionServerIT.scala:8)
	at org.specs2.specification.AroundEach$$anonfun$aroundContext$1$$anon$4.around(EachContext.scala:65)
	at org.specs2.specification.Around$class.apply(Context.scala:138)
	at org.specs2.specification.AroundEach$$anonfun$aroundContext$1$$anon$4.apply(EachContext.scala:65)
	at org.specs2.specification.create.ContextualFragmentFactory$$anonfun$example$3$$anonfun$apply$1.apply(FragmentFactory.scala:106)
	at org.specs2.specification.create.ContextualFragmentFactory$$anonfun$example$3$$anonfun$apply$1.apply(FragmentFactory.scala:106)
	at org.specs2.specification.core.Execution$$anonfun$execute$2$$anonfun$apply$4.apply(Execution.scala:73)
	at org.specs2.specification.core.Execution$$anonfun$execute$2$$anonfun$apply$4.apply(Execution.scala:73)
	at org.specs2.specification.core.Execution.setResult(Execution.scala:81)
	at org.specs2.specification.core.Execution$$anonfun$execute$2.apply(Execution.scala:73)
	at org.specs2.specification.core.Execution$$anonfun$execute$2.apply(Execution.scala:72)
	at scala.Option.fold(Option.scala:158)
	at org.specs2.specification.core.Execution.execute(Execution.scala:72)
	at org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1$$anonfun$apply$16.apply(Executor.scala:150)
	at org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1$$anonfun$apply$16.apply(Executor.scala:148)
	at org.specs2.specification.core.Fragment.updateExecution(Fragment.scala:49)
	at org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1.apply(Executor.scala:148)
	at org.specs2.specification.process.DefaultExecutor$$anonfun$executeFragment$1.apply(Executor.scala:147)
	at org.specs2.specification.process.DefaultExecutor$$anonfun$sequencedExecution$1$$anonfun$9.apply(Executor.scala:134)
	at org.specs2.specification.process.DefaultExecutor$$anonfun$sequencedExecution$1$$anonfun$9.apply(Executor.scala:134)
	at scalaz.concurrent.Task$.Try(Task.scala:456)
	at org.specs2.data.Processes$$anonfun$start$1.apply(Processes.scala:121)
	at org.specs2.data.Processes$$anonfun$start$1.apply(Processes.scala:121)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
	at scalaz.concurrent.Future$$anonfun$apply$15$$anon$3.call(Future.scala:432)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by etorreborre** [comment]
Thanks for reporting this issue, it is fixed now, I will publish a version later on today for you to try.

**Comment by etorreborre** [comment]
Ok, I released `4.0.2-958e92f-20171205091812` with the fix.
